src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtils.java

```diff
@@ -0,0 +1,176 @@
+        }
+
+        return response;
+    }
+
+    private static void handlePostponement(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getPostponementRequest().getPostponementRequestDetails();
+        String hearingVenue = sscsCaseData.getPostponementRequest().getPostponementRequestHearingVenue();
+        LocalDate hearingDate = LocalDateTime.parse(sscsCaseData.getPostponementRequest().getPostponementRequestHearingDateAndTime()).toLocalDate();
+        ADDITIONAL_REQUEST_DETAILS.append("Date of Hearing: ").append(hearingDate.format(DATE_TIME_FORMATTER)).append("\n");
+        ADDITIONAL_REQUEST_DETAILS.append("Hearing Venue: ").append(hearingVenue).append("\n");
+        ADDITIONAL_REQUEST_DETAILS.append("Reason for Postponement Request: ").append(requestDetails).append("\n");
+        title = "Postponement Request from FTA";
+    }
+
+    private static void handlePostHearing(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getDocumentGeneration().getBodyContent();
+        LocalDate issueFinalDecisionDate = sscsCaseData.getIssueFinalDecisionDate();
+        if (issueFinalDecisionDate == null) {
+            throw new IllegalArgumentException("issueFinalDecisionDate unexpectedly null for caseId: " + sscsCaseData.getCcdCaseId());
+        }
+        ADDITIONAL_REQUEST_DETAILS.append("Date of decision issued: ").append(issueFinalDecisionDate.format(DATE_TIME_FORMATTER)).append("\n");
+        String postHearingRequestType = sscsCaseData.getPostHearing().getRequestType().getDescriptionEn();
+        ADDITIONAL_REQUEST_DETAILS.append("Reason for ").append(postHearingRequestType).append(" request: ").append(requestDetails).append("\n");
+        title = String.format("%s Application from %s", postHearingRequestType, "FTA"); // TODO SSCS-10759 make dynamic for the uploading party
+    }
+
+    private static DocumentLink getPreviewDocument(
+        String pdfUrl,
+        String userAuthorisation,
+        GenerateFile generateFile,
+        String templateId,
+        String title
+    ) {
+        GenerateFileParams params = GenerateFileParams.builder()
+            .renditionOutputLocation(null)
+            .templateId(templateId)
+            .formPayload(PdfRequestTemplateBody.builder()
+                .title(title)
+                .text(ADDITIONAL_REQUEST_DETAILS.toString())
+                .build())
+            .userAuthentication(userAuthorisation)
+            .build();
+        final String generatedFileUrl = generateFile.assemble(params);
+
+        return DocumentLink.builder()
+            .documentFilename(pdfUrl)
+            .documentBinaryUrl(generatedFileUrl + "/binary")
+            .documentUrl(generatedFileUrl)
+            .build();
+    }
+
+    public static DocumentType getPostHearingDocumentType(SscsCaseData sscsCaseData) {
+        PostHearing postHearing = sscsCaseData.getPostHearing();
+        DocumentType documentType;
+
+        switch (postHearing.getRequestType()) {
```

Comment:
if else rather then switch unless you expect more scenarios here

Reply:
yes this is setting up for all the other request types (Correction etc)